pkg/cloudprovider/OWNERS

```diff
@@ -1,4 +1,43 @@
-assignees:
-  - mikedanese
-owners:
-  - mikedanese
+approvers:
+- mikedanese
+reviewers:
+- thockin
```

Comment:
These lists of reviewers look excessive, they include people who don't work often in pkg/cloudprovider or are no longer with the project